### ISSUE TITLE
When enabling EFI in harvester VM will not boot (no bootable device) Windows 11

### ISSUE THREAD

**Issue by avramadam** [harvester/harvester#5468]
[supportbundle_30f7398e-5270-42f2-87a9-48876053a0c5_2024-03-26T18-07-04Z.zip](https://github.com/harvester/harvester/files/14763176/supportbundle_30f7398e-5270-42f2-87a9-48876053a0c5_2024-03-26T18-07-04Z.zip)

**Comment by bk201** [comment]
@avramadam Thanks for the support bundle. Can you provide more steps on how to reproduce the issue?

**Comment by bagutzu** [comment]
@bk201 I had the same issue with Windows 11. It requires TPM/UEFI and Secure Boot to be enabled otherwise you cannot install it (or have to do some nasty [workarounds](https://jmcglock.substack.com/p/how-to-run-windows-11-on-a-harvester)). To reproduce it is easy, just use the windows-iso template from harvester and try to use a Windows 11 image with UEFI/TPM/Secure Boot Enabled and it will result in a "no bootable device".

We had to go with Windows 10 in the end (no UEFI/TPM/Secure Boot hard requirements).

**Comment by bk201** [comment]
@bagutzu Thanks for the update. But please allow me to double-check, do you mean (1) enabling TPM/UEFI and Secure boot with a Windows 11 ISO, but it can't boot the installer, or (2) the installer works and can do install, but the system can't boot after the installation completes and reboots?

**Comment by bagutzu** [comment]
You cannot boot into the installer. You can only boot into the installer with these features disabled.

**Comment by avramadam** [comment]
yes, #1, Linux installer boots fine with EFI enabled but not Windows.

**Comment by dimepues** [comment]
This is how you create a Windows 11 VM with TPM and SecureBoot.

1. use the Windows iso template but do NOT click create once you have applied your configurations.

2. Go to Advanced Options and click Windows under the OS Stratagy > OS Type drop down

<img width="1134" alt="Screenshot 2024-04-30 at 12 22 15 PM" src="https://github.com/harvester/harvester/assets/5528680/47c88375-3f89-4889-9afd-41a8d1b8f8c2">

4. Now scroll all the way to the bottom of the Advanced Options section. You will have an option to Enable TPM. Check the box to Enable TPM. Now the "Booting in EFI mode" option pops up.  Check the "Booting in EFI mode" option and Secure Boot option pops up. You can now create your Windows 11 VM.

<img width="706" alt="Screenshot 2024-04-30 at 12 22 29 PM" src="https://github.com/harvester/harvester/assets/5528680/70c42e24-613b-49ef-9221-9b4afdb59936">

<img width="706" alt="Screenshot 2024-04-30 at 12 22 40 PM" src="https://github.com/harvester/harvester/assets/5528680/07641c87-35c5-48df-8f63-7e0bfbb0570c">

Both Windows 11 and WIndows Server 2022 have worked for me using retail ISOs burned to USB using RuFus without customizations. The only thing that doesn't work:

- Memory Integrity
And not just because the of the E1000 drivers - once you get past that it just fails to configure)

- BitLocker auto-unlock w/ TPM. 
It always keeps prompting me for the recovery key even though I have reset the TPM protector and readded the key to the TPM.

**Comment by PhanLe1010** [comment]
I am hitting same issue as 

> @bk201 I had the same issue with Windows 11. It requires TPM/UEFI and Secure Boot to be enabled otherwise you cannot install it (or have to do some nasty [workarounds](https://jmcglock.substack.com/p/how-to-run-windows-11-on-a-harvester)). To reproduce it is easy, just use the windows-iso template from harvester and try to use a Windows 11 image with UEFI/TPM/Secure Boot Enabled and it will result in a "no bootable device".
> 
> We had to go with Windows 10 in the end (no UEFI/TPM/Secure Boot hard requirements).

![image](https://github.com/user-attachments/assets/e420faac-90c8-483f-8f86-b4353b8796cd)

**Comment by agowa** [comment]
Note, for whatever reason the UEFI boot of the iso takes way longer than the bios boot. It may be stuck at this screen here for 20 minutes before proceeding:
<img width="1610" height="1168" alt="Image" src="https://github.com/user-attachments/assets/f36f4447-eede-4253-b716-563498565b26" />
